static/panes/editor.js

```diff
@@ -267,67 +267,76 @@ Editor.prototype.initCallbacks = function () {
-    var paneAdderDropdown = this.domRoot.find('.add-pane');
+    scopedThis.topBar = scopedThis.domRoot.find('.top-bar');
+    scopedThis.hideable = scopedThis.domRoot.find('.hideable');
+
+    scopedThis.loadSaveButton = scopedThis.domRoot.find('.load-save');
+    var paneAdderDropdown = scopedThis.domRoot.find('.add-pane');
+    var addCompilerButton = scopedThis.domRoot.find('.btn.add-compiler');
+    scopedThis.conformanceViewerButton = scopedThis.domRoot.find('.btn.conformance');
+    var addEditorButton = scopedThis.domRoot.find('.btn.add-editor');
 
     var togglePaneAdder = function () {
         paneAdderDropdown.dropdown('toggle');
     };
-    this.topBar = this.domRoot.find('.top-bar');
 
     // NB a new compilerConfig needs to be created every time; else the state is shared
     // between all compilers created this way. That leads to some nasty-to-find state
     // bugs e.g. https://github.com/mattgodbolt/compiler-explorer/issues/225
     var getCompilerConfig = _.bind(function () {
-        return Components.getCompiler(this.id, this.currentLanguage.id);
+        return Components.getCompiler(scopedThis.id, scopedThis.currentLanguage.id);
     }, this);
 
     var getConformanceConfig = _.bind(function () {
-        return Components.getConformanceView(this.id, this.getSource());
+        return Components.getConformanceView(scopedThis.id, scopedThis.getSource());
     }, this);
 
-    this.container.layoutManager
-        .createDragSource(addCompilerButton, getCompilerConfig)
-        ._dragListener.on('dragStart', togglePaneAdder);
+    var getEditorConfig = _.bind(function () {
+        return Components.getEditor();
+    }, this);
 
-    addCompilerButton.click(_.bind(function () {
-        var insertPoint = this.hub.findParentRowOrColumn(this.container) ||
-            this.container.layoutManager.root.contentItems[0];
-        insertPoint.addChild(getCompilerConfig);
-    }, this));
+    function addDragListener(dragSource, dragConfig) {
+        scopedThis.container.layoutManager
+            .createDragSource(dragSource, dragConfig)
+            ._dragListener.on('dragStart', togglePaneAdder);
+    }
 
-    this.conformanceViewerButton = this.domRoot.find('.btn.conformance');
+    addDragListener(addCompilerButton, getCompilerConfig);
+    addDragListener(scopedThis.conformanceViewerButton, getConformanceConfig);
+    addDragListener(addEditorButton, getEditorConfig);
 
-    this.container.layoutManager
-        .createDragSource(this.conformanceViewerButton, getConformanceConfig)
-        ._dragListener.on('dragStart', togglePaneAdder);
-    this.conformanceViewerButton.click(_.bind(function () {
-        var insertPoint = this.hub.findParentRowOrColumn(this.container) ||
-            this.container.layoutManager.root.contentItems[0];
-        insertPoint.addChild(getConformanceConfig);
-    }, this));
+    function bindClickEvent(dragSource, dragConfig) {
+        dragSource.click(_.bind(function () {
```

Comment:
Here `this` and `scopedThis` are different.  And also in the below `keydown` event listener. 

I think `scopedThis` is a bad name, and probably something like `currentEditor` or `_editor` would be more appropriate.